### PR TITLE
Migrate to Capacitor 4

### PR DESCRIPTION
**Pull request by Nevvulo**
Updates the plugin to support Capacitor 4.
I've confirmed this is working for Android and iOS, tested in the simulator and through `npm run verify`

## Capacitor 3 vs Capacitor 4

- Version 1.0.3 (and lower) will target Capacitor 3
- Version 2.0.0 will target Capacitor 4

## Breaking Changes
* Version 1.x.x will solely support Capacitor 4
* No functionality has changed

### CODE REVIEW

File: android/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,6 +1,3 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (anonimitoraf) [inline android/src/main/AndroidManifest.xml:1]: Why was the package name changed? And does this have downstream impact?

File: ios/Podfile [hunk 2]
```diff
@@ -14,5 +14,4 @@ end
 
 target 'PluginTests' do
   capacitor_pods
-  pod 'Intercom', '~> 11.0'
```
Comment (anonimitoraf) [inline ios/Podfile:17]: Why was this removed?

**Review by anonimitoraf — COMMENTED** [review]
Again, nice job on this! I've left some q's

File: ios/Podfile (hunk 2 above)
Comment (Nevvulo) [reply ios/Podfile:17]: This is simply for the 'PluginTests' target which we're not leveraging at the moment, this line has moved to the 'Plugin' target which is correct

File: android/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -1,6 +1,3 @@
...
-    package="com.getcapacitor.community.intercom.intercom">
-    <uses-permission android:name="android.permission.INTERNET"/>
```
Comment (anonimitoraf) [inline android/src/main/AndroidManifest.xml:3]: Sorry, I only noticed this removal. Are these permissions not needed anymore?

Comment (Nevvulo) [reply android/src/main/AndroidManifest.xml:3]: Interestingly I've had no issues with the plugin without this, but I'll add this back as we had this before. Thanks for picking this up

**Review by gugahoi — APPROVED** [review]
*Looks* ok but so many things got removed it's hard to tell with confidence

File: android/src/main/AndroidManifest.xml (hunk 1 above)
Comment (gugahoi) [reply android/src/main/AndroidManifest.xml:1]: Did it change?

Comment (anonimitoraf) [reply android/src/main/AndroidManifest.xml:1]: Sorry, I quoted the wrong line, but yep, the line below:
`com.getcapacitor.community.intercom.intercom`
changed to
`com.joinflux.flux.intercom`

**Comment by Nevvulo** [comment]
The large number of deletions in this PR is from the removal of `example`, which is an example Angular Capacitor app that was automatically generated. The rest of the changes match pretty closely with the other migration PRs

File: android/src/main/AndroidManifest.xml (hunk 1 above)
Comment (Nevvulo) [reply android/src/main/AndroidManifest.xml:1]: >My main rationale was to ensure consistency across our plugins, and as for whether it has downstream impact I don’t think so. The versioning has changed to indicate that there are breaking changes and one of those would be the fact that the package name needs to be updated within `capacitor.plugins.json`, which happens automatically when updating the plugin and syncing. It’s mainly used to locate where the class is for the plugin on Android

**Review by gugahoi — APPROVED** [review]
